### PR TITLE
[FE/BE] wire up shelter login

### PR DESCRIPTION
**Pull request by stonefruit**
Changes:
- changed to use email instead of username as login id
- added constraints to user migrations
- redirect from /shelter/login to /shelter/home when login or register successful
- redirect to /shelter/login when Sign out at top right corner is clicked
 
To test:
- drop db, create db, `npm run db:migrate` in backend
- `npm run dev` in frontend
- `npm start` in backend
- visit http://localhost/shelter/login
- register, login and logout

### CODE REVIEW

File: backend/src/app.ts [hunk 1]
```diff
@@ -84,7 +85,7 @@ if (process.env.NODE_ENV === "development") {
 	app.use("/docs", express.static(__dirname + "/../../docs/"));
 }
 
-app.use("/", proxy(config.frontendUrl));
+app.use("/", checkPageAuth, proxy(config.frontendUrl));
```
Comment (stonefruit) [inline backend/src/app.ts:88]: handle Next page auth separate from api endpoints to make it less complex

File: backend/src/config/passport.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ const authStrategy = (passport: passport.PassportStatic): void => {
 		"local-signup",
 		new LocalStrategy(
 			{
-				usernameField: "username",
+				usernameField: "email",
```
Comment (stonefruit) [inline backend/src/config/passport.ts:32]: Email makes more sense as unique identifier. This changes the key passport uses as the identifier

File: backend/src/helpers/checkPageAuth.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+export default function checkPageAuth(
+	req: Request,
+	res: Response,
+	next: NextFunction
+): void {
+	const isAuthenticated = req.isAuthenticated();
+	const originalUrl = req.originalUrl;
+
+	if (isAuthenticated && originalUrl === "/shelter/login") {
```
Comment (stonefruit) [inline backend/src/helpers/checkPageAuth.ts:11]: This redirect helps for initial document GET when visiting the page.

File: backend/src/routes/auth.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ const authRouteSetup = (app: express.Application, passport: passport.PassportSta
     passport.authenticate('local-signup', (error: Error, user: User, info) => {
       if (error !== null) {
         const result = {"status": "failed", "message": "Failed to register"};
-        res.end(JSON.stringify(result));
+        return res.status(400).send(JSON.stringify(result));
```
Comment (stonefruit) [inline backend/src/routes/auth.ts:12]: Provide error status for ease of handling in UI

File: backend/src/routes/auth.ts [hunk 5]
```diff
@@ -44,7 +44,7 @@ const authRouteSetup = (app: express.Application, passport: passport.PassportSta
     })(req, res, next);
   });
 
-  app.get('/api/logout', (req: express.Request, res: express.Response) => {
+  app.post('/api/logout', (req: express.Request, res: express.Response) => {
```
Comment (stonefruit) [inline backend/src/routes/auth.ts:47]: https://stackoverflow.com/questions/3521290/logout-get-or-post

File: frontend/api/createAxiosInstance.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import axios from "axios";
+
+const createAxiosInstance = () => {
```
Comment (stonefruit) [inline frontend/api/createAxiosInstance.ts:3]: Needs to be called at runtime as Nextjs tries to render `window.location.origin` in the Next server when there is no window

File: frontend/components/shelter/login/LoginForm.tsx [hunk 7]
```diff
@@ -45,27 +47,27 @@ const LoginForm = ({ onFinish }: LoginFormProps) => (
 				placeholder="Password"
 			/>
 		</Form.Item>
-		<Form.Item>
+		{/* <Form.Item>
```
Comment (stonefruit) [inline frontend/components/shelter/login/LoginForm.tsx:50]: Hid a bunch of thing we wont be using yet

File: frontend/components/shelter/login/SignUpForm.tsx [hunk 8]
```diff
@@ -42,16 +43,16 @@ const SignUpForm = ({ onFinish }: SignUpFormProps) => (
 		</Form.Item>
 		<Form.Item>
 			<Button type="primary" htmlType="submit">
-				Login
+				Sign Up
 			</Button>
 		</Form.Item>
-		<Form.Item>
+		{/* <Form.Item>
```
Comment (stonefruit) [inline frontend/components/shelter/login/SignUpForm.tsx:49]: Hid a bunch of thing we wont be using yet

File: frontend/types.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+export type LoginFormValues = {
```
Comment (stonefruit) [inline frontend/types.ts:1]: Moved here for explicit import of type, which is more obvious.

**Review by ryhuz — APPROVED** [review]

**Pull request merged**